### PR TITLE
Update week3 func/non-func requirements

### PR DESCRIPTION
**Pull request by ChrBank**
Please check if you agree with the requirements, whether they are functional or non-functional requirements. In the non-functional requirement, I am a bit confused with the second, the third, and the forth.

### CODE REVIEW

File: docs/doc.md [hunk 1]
```diff
@@ -49,8 +49,22 @@
...
+
+## Week 3
+1. The REST API shall receive a repository identifier from GitHub. The form could be:
+    1. <github_user>/<repository_name> 
+    2. <github_organization>/<repository_name>
+2. If the repository does not exist locally, then your GitInsight application shall clone the remote repository from GitHub and store it in a temporary     local directory on your computer.
+3. In case the repository was already cloned earlier, then the respective local repository shall be updated. That is, using libgit2sharp your application should update the local repository similar to running a git pull if you were to update a Git repository manually.
+4. The analyses that your GitInsight application is performing on that now cloned local repository remain the same.
+5. The REST API shall return the analysis results via a JSON objects.
+6. Illustrate with an UML activity diagram the sequence of operations that your GitInsight application performs once triggered with a respective GET request and until it responds with the corresponding JSON data.
```
Comment (fredpetersen) [inline docs/doc.md:70]: not a requirement for the product, but rather for documentation (which isn't covered by these requirements)

File: docs/doc.md [hunk 2]
```diff
@@ -49,8 +49,22 @@
...
 3. Has to implement resonable test.
 
+## Week 3
+1. Implement one or more integration tests in your test suite. 
+2. Illustrate the architecture of your REST service using a suitable notation.
+
 # Non Functional Requirements
 ## Week 1
 None
 ## Week 2
```
Comment (fredpetersen) [inline docs/doc.md:59]: Perhaps we should change the requirements for earlier weeks as well (since we know they're non-functional now) while we're at it

**Review by fredpetersen — CHANGES_REQUESTED** [review]
Some changes

**Review by Lucasfth — APPROVED** [review]
lgtm

**Review by fredpetersen — APPROVED** [review]
good

**Pull request merged**